tests/unit/additives.t

```diff
@@ -7,992 +7,625 @@ use Test2::V0;
+		}
+	]
 );
 
-is($product_ref->{additives_original_tags}, ['en:e330', 'en:e120', 'en:e500',],);
-
-$product_ref = {
-	lc => "fi",
-	ingredients_text =>
-		"Sian rinta, suola, säilöntäaineet (kaliumlaktaatti), natriumnitriitti, luontainen aromi, glukoosisiirapppi, hapettumisenestoaine (natriumerytorbaatti)"
-};
-
-extract_additives_from_text($product_ref);
-
-is($product_ref->{additives_original_tags}, ['en:e326', 'en:e250', 'en:e316',],);
-
-is(canonicalize_taxonomy_tag("fi", "additives", "natriumerytorbaatti"), "en:e316");
-is(canonicalize_taxonomy_tag("fi", "additives", "sitruunahappo"), "en:e330");
-
-$product_ref = {
-	lc => "fi",
-	ingredients_text =>
-		"sakeuttamisaine arabikumi, makeutusaineet (sorbitoli, maltitolisiirappi, asesulfaami K), happamuudensäätöaine sitruunahappo, pintakäsittelyaine mehiläisvaha"
-};
-
-extract_additives_from_text($product_ref);
-
-is($product_ref->{additives_original_tags}, ['en:e414', 'en:e420', 'en:e965ii', 'en:e950', 'en:e330', 'en:e901',],);
+foreach my $test_ref (@tests) {
+	my $testid = $test_ref->[0];
+	my $product_ref = $test_ref->[1];
+
+	# Run the test
+	extract_additives_from_text($product_ref);
+
+	# If there's a language test, perform it
+	if (exists $product_ref->{ingredients_text_fr}) {
+		compute_languages($product_ref);
+		clean_ingredients_text($product_ref);
+		extract_additives_from_text($product_ref);
+	}
+
+	# Remove any temporary OCR fields that might cause test failures
```

Comment:
I don't understand how there could be OCR fields, as all the inputs are specified in this additives.t file, and they don't have OCR fields.

Reply:
ah ok found it, it can be added by clean_ingredients_text()